hugging_face_classifier_trainer/hugging_face_classifier_trainer.py

```diff
@@ -0,0 +1,794 @@
+    dataset: DataItem = None,
+    drop_columns: Optional[List[str]] = None,
+    pretrained_tokenizer: str = None,
+    pretrained_model: str = None,
+    model_class: str = None,
+    model_name: str = "huggingface-model",
+    label_name: str = "labels",
+    text_col: str = "text",
+    num_of_train_samples: int = None,
+    train_test_split_size: float = None,
+    metrics: List[str] = None,
+    random_state: int = None,
+):
+    """
+    Training and evaluating a pretrained model with a pretrained tokenizer over a dataset.
+    The dataset can be either be the name of the dataset that contains in the HuggingFace hub,
+    or a URI or a FeatureVector
+
+    :param context:                 MLRun context
+    :param hf_dataset:              The name of the dataset to get from the HuggingFace hub
+    :param dataset:                 The dataset to train the model on. Can be either a URI or a FeatureVector
+    :param drop_columns:            The columns to drop from the dataset.
+    :param pretrained_tokenizer:    The name of the pretrained tokenizer from the HuggingFace hub.
+    :param pretrained_model:        The name of the pretrained model from the HuggingFace hub.
+    :param model_name:              The model's name to use for storing the model artifact, default to 'model'
+    :param model_class:             The class of the model, e.g. `transformers.AutoModelForSequenceClassification`
+    :param label_name:              The target label of the column in the dataset.
+    :param text_col:                The input text column un the dataset.
+    :param num_of_train_samples:    Max number of training samples, for debugging.
+    :param train_test_split_size:   Should be between 0.0 and 1.0 and represent the proportion of the dataset to include
+                                    in the test split.
+    :param metrics:                 List of different metrics for evaluate the model such as f1, accuracy etc.
+    :param random_state:            Random state for train_test_split
+    """
+
+    if not label_name or not pretrained_tokenizer:
+        raise mlrun.errors.MLRunRuntimeError(
+            "Must provide label_names and pretrained_tokenizer"
+        )
+
+    if train_test_split_size is None:
+        context.logger.info(
+            "test_set or train_test_split_size are not provided, setting train_test_split_size to 0.2"
+        )
+        train_test_split_size = 0.2
+
+    # Creating tokenizer:
+    tokenizer = AutoTokenizer.from_pretrained(pretrained_tokenizer)
+
+    def preprocess_function(examples):
+        return tokenizer(examples[text_col], truncation=True)
+
+    # prepare data for training
+    if hf_dataset:
+        train_dataset, test_dataset = _prepare_dataset(
+            context,
+            hf_dataset,
+            label_name,
+            drop_columns,
+            num_of_train_samples,
+            train_test_split_size,
+            random_state=random_state,
+        )
+    elif dataset:
```

Comment:
if both `hf_dataset` and `dataset` are not given? You should catch it with an `else`